code/modules/mob/living/silicon/robot/gripper.dm

```diff
@@ -63,60 +59,108 @@
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(I.loc == (R || R.module))// Don't remove your own modules
+			to_chat(R, "<span class='danger'>ERROR. Safety protocols prevent self-disassembling.</span>")
+			return FALSE
 	if (!wrapped)
-		if(is_type_in_list(I,can_hold))
-			if (feedback)
-				to_chat(user, "You collect \the [I].")
+		var/grab = FALSE
+		for(var/typepath in can_hold)
+			if(istype(I,typepath))
+				grab = TRUE
+				break
+		if(grab && !is_type_in_list(I, blacklist))
+			if(feedback)
+				to_chat(user, "<span class='notice'>You collect \the [I].</span>")
 			I.do_pickup_animation(user.loc, I.loc)
-			I.forceMove(src)
+			I.loc = src
+			I.add_fingerprint(user)
 			wrapped = I
 			update_icon()
 			return TRUE
-		if (feedback)
-			to_chat(user, "<span class='danger'>Your gripper cannot hold \the [I].</span>")
+		if(feedback)
+			to_chat(user, "<span class='danger'>ERROR. Your [name] wasn't designed to handle \the [I].</span>")
 		return FALSE
-	if (feedback)
-		to_chat(user, "<span class='danger'>Your gripper is already holding \the [wrapped].</span>")
+	if(feedback)
+		to_chat(user, "<span class='danger'>ERROR. Your [name] is already holding \the [wrapped].</span>")
 	return FALSE
 
 
 //This places a little image of the gripped item in the gripper, so you can see visually what you're holding
 /obj/item/gripper/update_icon()
-	underlays.Cut()
-	if (wrapped && wrapped.icon)
-		var/mutable_appearance/MA = new(wrapped)
-		MA.layer = ABOVE_HUD_LAYER
-		MA.plane = get_relative_plane(ABOVE_HUD_PLANE)
-
-		//Reset pixel offsets to initial values, incase being on a table messed them up
-		//And then subtract 8 from the Y value so it appears in the claw at the bottom
-		MA.pixel_y = initial(MA.pixel_y)-8
-		MA.pixel_x = initial(MA.pixel_x)
+	overlays.Cut()
+	if(wrapped && wrapped.icon)
+		var/image/olay = image("icon" = wrapped.icon, "icon_state" = wrapped.icon_state, "layer" = 30 + wrapped.layer, "pixel_x" = null, "pixel_y" = null)
+		olay.overlays = wrapped.overlays
+		olay.appearance_flags = RESET_ALPHA
+		alpha = 128
+		overlays += olay
+	else
+		alpha = initial(alpha)
+	..()
 
-		underlays += MA
+/obj/item/gripper/AltClick()
+	if(gripper_sanity_check(src))
+		.=wrapped.AltClick()
+		update_icon()
+		return
+	return ..()
 
-/obj/item/gripper/attack_self(mob/user as mob)
-	if(wrapped)
-		.=wrapped.attack_self(user)
+/obj/item/gripper/CtrlClick()
+	if(gripper_sanity_check(src))
+		.=wrapped.CtrlClick()
 		update_icon()
 		return
 	return ..()
 
-/obj/item/gripper/AltClick(mob/user as mob)
-	if(wrapped)
-		.=wrapped.AltClick(user)
+/obj/item/gripper/attack_self(var/mob/living/user)
+	if(gripper_sanity_check(src))
+		.=wrapped.attack_self(user)
 		update_icon()
+		return
 	return ..()
 
-/obj/item/gripper/verb/drop_item()
+/obj/item/gripper/proc/drop_item(var/obj/item/to_drop, var/atom/target, force_drop = 0,var/dontsay = null) //Do we care about to_drop at all?
```

Comment:
hit Q to drop on floor, or click on table to place!
(you will screwdriver tables, so use a toolbox you fucking gremlin)